numpy/lib/arraypad.py

```diff
@@ -32,13 +32,13 @@ def _linear_ramp(ndim, axis, start, stop, size, reverse=False):
         the one specified by `axis` will have the size 1.
     axis : int
         The dimension that contains the linear ramp of `size`.
-    start : int or ndarray
+    start : int or float or ndarray
```

Comment:
Perhaps just `Number` instead of naming `float` and `int` - arrays of `Decimal` should work here too, and the numpy complex types I'd also expect to work.